py/desitarget/cuts.py

```diff
@@ -94,22 +94,70 @@ def isFSTD_colors(gflux=None, rflux=None, zflux=None, w1flux=None, w2flux=None,
     Returns:
         mask : boolean array, True if the object has colors like an FSTD
 
+    """
+
+    if primary is None:
+        primary = np.ones_like(gflux, dtype='?')
+    fstd = primary.copy()
+
+    # Clip to avoid warnings from negative numbers.
+    gflux = gflux.clip(0)
```

Comment:
Lines 104 to 110 look dangerous. For example, If `rflux < 0` then `rflux = 0` and then `np.log10(rflux / flux)` is -inf and potentially flags a RuntimeWarning. We probably want to clip to a number ever-so-slightly larger than 0? But then `rflux / gflux` is 1 and `grcolor` and `rzcolor` would incorrectly be 0. Is there a way to elegantly catch zero flux without having 0 color?